### PR TITLE
Add Worker#subscribes_to for defining routing_key without inferring from class name

### LINKED ISSUE #65: Add ability to use conventions but make namespace/class name more semantic
right now we do:
```ruby
class OrderCompleted
  include LaGear::Worker

  def perform(order_id)
    # do work to send a complimentary slap bracelet
  end
end
```

when it should actually be
```ruby
class SlapBraceletSender
  include LaGear::Worker

  def perform(order_id)
    # do work to send a complimentary slap bracelet
  end
end
```

and then somehow automatically configuring the routing key queue name without having to do manually like in Sneakers.

### PR DESCRIPTION
**Pull request by gabrieljoelc**
Closes #65 

This is backwards compatible but will now allow you to define the routing key but still set up the queue name and retry routing keys for you:
```ruby
Sneakers::CONFIG[:app_name] = 'bo_knows'

module SlapBraceletSender
  class V1
    include LaGear::Worker
  
    subscribes_to 'slap_bracelet_created.v2'

    def perform(order_id)
      # do work to send a complimentary slap bracelet
    end
  end
end
```
Here's what you would get:
- Routing key - `slap_bracelet_created.v2`
- Queue name - `bo_knows.slap_bracelet_sender.v1`
- Retry routing key - `bo_knows.slap_bracelet_sender.v1.retry`

### CODE REVIEW

**Comment by gabrieljoelc** [comment]
One caveat to this, is routing keys will not automatically version-increment when a handler's version increments. It will need to be done manually. If the `slap_bracelet_created.v2` message type increments to `slap_bracelet_created.v3`, you will have to create `SlapBraceletSender::V2` and invoke ``subscribes_to 'slap_bracelet_created.v3'`` in the new worker.

File: lib/la_gear/worker.rb [hunk 1]
```diff
@@ -53,6 +53,12 @@ def retry_routing_key
       def default_queue_args
         { 'x-dead-letter-exchange' => "#{::Sneakers::CONFIG.fetch(:exchange, 'sneakers').underscore}.retry" }
```
Comment (crazydog115) [inline lib/la_gear/worker.rb]: Can we pull this out into a method
`#{::Sneakers::CONFIG.fetch(:exchange, 'sneakers').underscore`

Comment (gabrieljoelc) [inline lib/la_gear/worker.rb]: what's the value?

Comment (crazydog115) [inline lib/la_gear/worker.rb]: Oh I actually meant the other one that's used twice `::Sneakers::CONFIG.fetch(:app_name, 'sneakers').underscore` in `default_queue_name` and `subscribes_to`

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,4 @@ tmp
 *.a
 mkmf.log
 *.log
+/.idea
```
Comment (gabrieljoelc) [inline .gitignore]: newline

**Pull request merged**